Engine/BaseAssets/Components/Postprocessing/Bloom/PixelShaderBloom.cs

```diff
@@ -0,0 +1,205 @@
+﻿using LinearAlgebra;
+using SharpDX;
+using SharpDX.DXGI;
+using SharpDX.Direct3D11;
+using System;
+using System.Windows.Documents;
+using System.Collections.Generic;
+
+namespace Engine.BaseAssets.Components.Postprocessing.Bloom
+{
+    public class PixelShaderBloom : PostProcessEffect
+    {
+        private const int MaxIterationsCount = 16;
+        private SharpDX.Direct3D11.Device device => GraphicsCore.CurrentDevice;
+
+        private Texture[] samplingsTextures;
+        private Sampler sampler;
+        private RasterizerState backCullingRasterizerState;
+        private BlendState additiveBlendState;
+
+        private ShaderPipeline downsampleBoxPipeline;
+        private ShaderPipeline upsampleBoxPipeline;
+        private ShaderPipeline prefilterPipeline;
+
+        private int samplingsCount;
+
+        public PixelShaderBloom(int screenWidth, int screenHeight, int iterationsCount = 5)
+        {
+            if(iterationsCount <= 0 && iterationsCount > MaxIterationsCount)
+                throw new ArgumentOutOfRangeException(nameof(iterationsCount));
+
+            samplingsCount = iterationsCount;
+
+            sampler = new Sampler(TextureAddressMode.Clamp, TextureAddressMode.Clamp, Filter.MinMagMipLinear);
```

Comment:
if sampler is the same for each bloom effect copy - make it private static field and initialize it in static constructor, same goes for rasterizer and blend state (and remind me to do the same for particle effects shaders later)